network/p2p/scoring/score_option.go

```diff
@@ -150,33 +164,28 @@ func (s *ScoreOption) preparePeerScoreThresholds() {
 
 // preparePeerScoreParams prepares the peer score parameters for the pubsub system.
 // It is based on the default parameters defined in libp2p pubsub peer scoring.
-func (s *ScoreOption) preparePeerScoreParams() {
-	s.peerScoreParams = &pubsub.PeerScoreParams{
-		// we don't set all the parameters, so we skip the atomic validation.
-		// atomic validation fails initialization if any parameter is not set.
-		SkipAtomicValidation: true,
-
-		// DecayInterval is the interval over which we decay the effect of past behavior. So that
-		// a good or bad behavior will not have a permanent effect on the score.
-		DecayInterval: time.Hour,
-
-		// DecayToZero defines the maximum value below which a peer scoring counter is reset to zero.
-		// This is to prevent the counter from decaying to a very small value.
-		// The default value is 0.01, which means that a counter will be reset to zero if it decays to 0.01.
-		// When a counter hits the DecayToZero threshold, it means that the peer did not exhibit the behavior
-		// for a long time, and we can reset the counter.
-		DecayToZero: 0.01,
-
-		// AppSpecificScore is a function that takes a peer ID and returns an application specific score.
-		// At the current stage, we only use it to penalize and reward the peers based on their subscriptions.
-		AppSpecificScore: s.appSpecificScoreFunction,
-		// AppSpecificWeight is the weight of the application specific score.
-		AppSpecificWeight: DefaultAppSpecificScoreWeight,
-	}
+func (s *ScoreOption) defaultPeerScoreParams() {
+	// we don't set all the parameters, so we skip the atomic validation.
+	// atomic validation fails initialization if any parameter is not set.
+	s.peerScoreParams.SkipAtomicValidation = true
+	// DecayInterval is the interval over which we decay the effect of past behavior. So that
+	// a good or bad behavior will not have a permanent effect on the score.
+	s.peerScoreParams.DecayInterval = time.Hour
+	// DecayToZero defines the maximum value below which a peer scoring counter is reset to zero.
+	// This is to prevent the counter from decaying to a very small value.
+	// The default value is 0.01, which means that a counter will be reset to zero if it decays to 0.01.
+	// When a counter hits the DecayToZero threshold, it means that the peer did not exhibit the behavior
+	// for a long time, and we can reset the counter.
+	s.peerScoreParams.DecayToZero = 0.01
+	// AppSpecificScore is a function that takes a peer ID and returns an application specific score.
+	// At the current stage, we only use it to penalize and reward the peers based on their subscriptions.
+	s.peerScoreParams.AppSpecificScore = s.appSpecificScoreFunction
+	// AppSpecificWeight is the weight of the application specific score.
+	s.peerScoreParams.AppSpecificWeight = DefaultAppSpecificScoreWeight
 }
 
 func (s *ScoreOption) BuildGossipSubScoreOption() pubsub.Option {
-	s.preparePeerScoreParams()
+	s.defaultPeerScoreParams()
```

Comment:
Elsewhere in the codebase we have a method that returns a new object/slice containing default config. e.g.
https://github.com/onflow/flow-go/blob/f7da574c2ce1544e79eef17c85210fd0f5a8ec2a/cmd/node_builder.go#L277

Can we use that same pattern here for `defaultPeerScoreParams` and `preparePeerScoreThresholds`?

Reply:
https://github.com/onflow/flow-go/pull/3986/commits/62f9ec3af5c205338ea335acf5646cad8705a5a6